sync/src/synchronizer/headers_process.rs

```diff
@@ -393,49 +393,74 @@ where
 
     pub fn accept(&self) -> ValidationResult {
         let mut result = ValidationResult::default();
+
+        // FIXME If status == BLOCK_INVALID then return early. But which error
```

Comment:
Use a separate error code such as DuplicatedInvalidBlock or save the invalid reason as well.

Reply:
I would like to persist a `recent_reject { hash => reason }` to record the invalid reasons.